skglm/solvers/group_bcd_solver.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from numba import njit
+
+
+def grp_bcd_solver(X, y, datafit, penalty,
+                   max_iter=1000, tol=1e-7, verbose=False):
+    """Run a group BCD solver.
+
+    Parameters
+    ----------
+    X : array, shape (n_samples, n_features)
+        Design matrix.
+
+    y : array, shape (n_samples,)
+        Target vector.
+
+    datafit : QuadraticGroup
+        DataFit object.
+
+    penalty : WeightedGroupL1
+        Penalty object.
+
+    max_iter : int, default 1000
+        Maximum number of iterations.
+
+    tol : float, default 1e-7
+        Tolerance for convergence.
+
+    verbose : bool, default False
+        Log or not the objective at each iteration.
+
+    Returns
+    -------
+    w : array, shape (n_features,)
+        Vector that minimizes ``datafit() + penalty()``.
+    """
+    n_samples, n_features = X.shape
+    n_groups = len(penalty.grp_partition) - 1
```

Comment:
maybe check that penalty has the attributes needed for a group, and raise a ValueError otherwise ? your call